File: screenSplash/app/src/main/java/com/example/splashscreen/SplashFragment.java
package com.example.splashscreen;

import android.os.Bundle;
import android.os.Handler;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;


public class SplashFragment extends Fragment {




    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {


        new Handler().postDelayed(new Runnable() {
            public void run() {
                Navigation.findNavController(getView()).navigate(R.id.viewPagerFragment);
            }
        }, 5000);




                return inflater.inflate(R.layout.fragment_splash, container, false);
    }
}

File: MyApplication/app/src/main/java/com/example/myapplication/splashFragment.kt
package com.example.myapplication

import android.content.Context
import android.os.Bundle
import android.os.Handler
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.fragment.findNavController

class splashFragment : Fragment() {
     override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
     Handler().postDelayed({
         if(onBoardingFinished())
         {
             findNavController().navigate(R.id.action_splashFragment_to_homeFragment)
         }
         else
         {
             findNavController().navigate(R.id.action_splashFragment_to_viewPagerFragment)
         }

     },3000)
        return inflater.inflate(R.layout.fragment_splash, container, false)
    }
    private fun onBoardingFinished():Boolean {
        val screenPref = requireActivity().getSharedPreferences("onBoarding", Context.MODE_PRIVATE)
    return screenPref.getBoolean("Finished",false )
    }
}